### DISCUSSION TITLE
How do I make custom variants stack-able with my responsive variants?

### DISCUSSION THREAD

**Discussion by JoshuaSoileau** [tailwindlabs/tailwindcss#4662] [Help]
[Demo](https://play.tailwindcss.com/YBxIfSKKwr?file=config) of what I'm trying to accomplish.

How do you make any `custom variants` you create stack-able with the `responsive` variants?
I want my new variant to be able to be changed responsively, but the `sm/md/lg` don't seem to stack with any new variant that I create.

I know you can apply the responsive variants to any new `utilities` that you create with `addUtility`, but how do I make sure my custom variants can be changed responsively?

**Comment by VicGUTT** [comment]
Hey !

Considering your demo, you made a small mistake when setting your custom variant to the scale property. By setting it like this :

```js
variants: { scale: ['custom-checked'] }
```

you are overwriting all the default variants and so **only** "custom-checked" variants will be generated.
To prevent that from happening you could either add "responsive" to the array :

```js
variants: { scale: ['responsive', 'custom-checked'] }
```

or extend the default variants :

```js
variants: {
  extend: {
    scale: ['custom-checked'],
  },
},
```

Although, extending didn't seem to work for some reason in the demo...

Hope it help ✌🏾

**Reply by JoshuaSoileau** [reply]
Thanks @VicGUTT 

Doesn't help with the problem, though, unfortunately.

**Reply by VicGUTT** [reply]
Hey, interesting, [tweaking your demo](https://play.tailwindcss.com/RPVFvkWqwX?file=config) does fix it for me.
I have also tested it locally, both with and without JIT enabled.

I might need more context then in order to be of help. But `'responsive'` does need to be part of the variant scale's array, otherwise responsive variants wont be created at all for scaling utilities.

Also, maybe infos like what Tailwind version are you using, in what environnement/tooling etc... may also be helpful.

**Comment by JoshuaSoileau** [comment]
I'm not sure what the github discussion etiquette is, but `bump`
